src/irmin-pack/unix/gc.ml

```diff
@@ -286,7 +289,10 @@ module Worker = struct
           >>= (fun _ -> Ao.close suffix)
           |> Errs.log_if_error "GC: Close suffix")
       @@ fun () ->
-      let read_exn = Dispatcher.read_exn dispatcher in
+      let read_exn ~off ~len buf =
+        let accessor = Dispatcher.create_accessor_exn dispatcher ~off ~len in
```

Comment:
maybe this is not a problem, but I think you are allocating more with this change? Since in this context the location is necessarily the suffix, there is maybe a way to prevent these allocations?

Reply:
I don't think this will be a problem. I think we've been a bit too pessimistic with the impact of these extra objects. We'll need a deeper analysis